packages/blade/src/components/Input/SearchInput/_decisions/decisions.md

```diff
@@ -0,0 +1,184 @@
+# SearchInput
+
+The Search Input component enables users to input text and find relevant information within a diverse range of data or choices.
+
+This document serves as an overview of the API for the `SearchInput` component.
+
+<img src="./searchinput-thumbnail.png" width="50%" alt="Thumbnail" />
+
+<img src="./searchinput-dropdown.png" width="50%" alt="Input with results in dropdown" />
+
+## Design
+
+- [Explore the design in Figma: SearchInput](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=85072%3A160345&mode=design&t=Pv93G8LK6OtL4wwk-1)
+
+## Anatomy
+
+<img src="./searchinput-anatomy.png" width="50%" alt="Anatomy" />
+
+## API
+
+The `SearchInput` component extends the `BaseInput` and shares common props:
+
+```ts
+type CommonProps = Pick<
+  BaseInputProps,
+  | 'label'
+  | 'labelPosition'
+  | 'name'
+  | 'value'
+  | 'helpText'
+  | 'isDisabled'
+  | 'onChange'
+  | 'onFocus'
+  | 'onBlur'
+  | 'onSubmit'
+>;
+
+type SearchInputProps = CommonProps & {
+  /**
+   * The size of the input field.
+   *
+   * @default medium
+   */
+  size?: 'medium' | 'large';
+  /**
+   * Toggles the loading state of the component
+   *
+   * @default true
+   */
+  isLoading?: boolean;
+  /**
+   * Callback that is called when the clear button is clicked.
+   */
+  onClearButtonClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+};
+```
+
+## Usage
+
+### With Dropdown
+
+> **Note**
+>
+> The `event` parameter in the ActionListItem's `onClick` prop needs to be exposed to work with `react-router` for navigation handling.
+
+```jsx
+import React from 'react';
+import {
+  Dropdown,
+  DropdownOverlay,
+  SearchInput,
+  ActionList,
+  ActionListItem,
+  Box,
+} from '@razorpay/blade/components';
+
+const App = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const menuItems = [
+    'Settings',
+    'Profile',
+    'Transactions',
+    'Help',
+    'Refunds',
+    'Settlements',
+    'Payouts',
+  ];
+  const popularItems = ['Transactions', 'Settlements'];
+
+  return (
+    <Dropdown>
+      <SearchInput
+        label="Search"
+        onChange={({ value }) => setSearchTerm(value)}
+        placeholder="Search here"
```

Comment:
Have we explored non-compound API here?

I see that items are being set from array only so we need data in array.

In SelectInput and AutoComplete, we went with this compound API because consumers can keep their data inside `<ActionListItem />` individually, add sections, descriptions, trailing text etc

If keeping data in anyway a requirement for SearchInput, can we just have-

```
<SearchInput
  label="Something"
  data={[

  ]}
/>
```

... type of API and internally use Dropdown component?

Reply:
Discussed on slack. SearchInput is small wrapper on top of TextInput. It also has modal and pattern-like usecases so better to not tie it to Dropdown.